packages/gen-ai-hub/src/core.ts

```diff
@@ -0,0 +1,71 @@
+import { DeploymentApi, AiDeployment } from '@sap-ai-sdk/ai-core';
+
+// TODO: docs
+/* eslint-disable */
+
+export type DeploymentResolver = DeploymentId | (() => Promise<AiDeployment>);
+export type DeploymentId = string;
+
```

Comment:
relevant context also [here](https://github.com/SAP/ai-sdk-js/pull/50#discussion_r1702004299) and [here](https://github.com/SAP/ai-sdk-js/pull/50#discussion_r1701879493)